### PR TITLE
Create User model from Devise, registration form. All tests passing.

### PR DESCRIPTION
**Pull request by AmberWilkie**
PT id: #130182033
link: https://www.pivotaltracker.com/n/projects/1863711/stories/130182033

Scaffold Rails app
Set up testing environment, including continuous integration & deployment (from dev branch)
Create user registration function with Devise

### CODE REVIEW

File: app/assets/javascripts/cable.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//= require action_cable
+//= require_self
+//= require_tree ./channels
+
+(function() {
+  this.App || (this.App = {});
+
+  App.cable = ActionCable.createConsumer();
+
+}).call(this);
```
Comment (tochman) [inline app/assets/javascripts/cable.js]: Delete this file

File: app/channels/application_cable/channel.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Channel < ActionCable::Channel::Base
+  end
+end
```
Comment (tochman) [inline app/channels/application_cable/channel.rb]: Delete this file

File: app/channels/application_cable/connection.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+module ApplicationCable
+  class Connection < ActionCable::Connection::Base
+  end
+end
```
Comment (tochman) [inline app/channels/application_cable/connection.rb]: Delete this file

File: app/controllers/registration_controller.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+class RegistrationController < ApplicationController
+  def index
+    @user = User.new
+  end
+end
```
Comment (tochman) [inline app/controllers/registration_controller.rb]: Delete this file

File: app/jobs/application_job.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+class ApplicationJob < ActiveJob::Base
+end
```
Comment (tochman) [inline app/jobs/application_job.rb]: Delete this file

File: config/cable.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+development:
+  adapter: async
+
+test:
+  adapter: async
+
+production:
+  adapter: redis
+  url: redis://localhost:6379/1
```
Comment (tochman) [inline config/cable.yml]: Delete this file

File: config/database.yml [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+#     url: <%= ENV['DATABASE_URL'] %>
+#
+production:
+  <<: *default
+  database: garden_club_production
+  username: garden_club
+  password: <%= ENV['GARDEN_CLUB_DATABASE_PASSWORD'] %>
+
+cucumber:
+  <<: *test
```
Comment (tochman) [inline config/database.yml:19]: Clear out comments

File: config/routes.rb [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Rails.application.routes.draw do
+  get 'home/index'
```
Comment (tochman) [inline config/routes.rb]: ??

File: config/routes.rb [hunk 9]
```diff
@@ -0,0 +1,8 @@
...
+
+  devise_for :users
+  get 'registration/index'
```
Comment (tochman) [inline config/routes.rb]: ??

File: config/routes.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+  resources :users
```
Comment (tochman) [inline config/routes.rb]: ??

File: config/routes.rb [hunk 11]
```diff
@@ -0,0 +1,8 @@
...
+  root to: "home#index"
```
Comment (tochman) [inline config/routes.rb]: root controller: :home, action: :index
or something like this...

File: db/migrate/20160913094721_add_name_to_users.rb [hunk 12]
```diff
@@ -0,0 +1,6 @@
+class AddNameToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :name, :string
```
Comment (tochman) [inline db/migrate/20160913094721_add_name_to_users.rb]:
```
add_column :users :name, :string, null: false, default: ""
```

File: db/migrate/20160913094721_add_name_to_users.rb [hunk 13]
```diff
@@ -0,0 +1,6 @@
...
+    add_column :users, :email_confirmation, :string
```
Comment (tochman) [inline db/migrate/20160913094721_add_name_to_users.rb]: delete this line

File: db/migrate/20160913095735_add_null_constraint_for_name_to_user.rb [hunk 14]
```diff
@@ -0,0 +1,5 @@
+class AddNullConstraintForNameToUser < ActiveRecord::Migration[5.0]
+  def change
+    change_column :users, :name, :string, null: false
+  end
+end
```
Comment (tochman) [inline db/migrate/20160913095735_add_null_constraint_for_name_to_user.rb]: Delete this file

File: db/migrate/20160913100759_add_default_value_to_name.rb [hunk 15]
```diff
@@ -0,0 +1,5 @@
+class AddDefaultValueToName < ActiveRecord::Migration[5.0]
+  def change
+    change_column :users, :name, :string, null: false, default: ""
+  end
+end
```
Comment (tochman) [inline db/migrate/20160913100759_add_default_value_to_name.rb]: Delete this file

File: spec/factories/users.rb [hunk 16]
```diff
@@ -0,0 +1,9 @@
+FactoryGirl.define do
+  factory :user do
+    name "My Name"
+    email "My@Email.com"
+    email_confirmation "My@Email.com"
```
Comment (tochman) [inline spec/factories/users.rb]: Remove

**Comment by AmberWilkie** [comment]
PT id: #130182033
link: https://www.pivotaltracker.com/n/projects/1863711/stories/130182033

Scaffold Rails app
Set up testing environment, including continuous integration & deployment (from dev branch)
Create user registration function with Devise

**Pull request merged**